build/gen_stub.php

```diff
@@ -387,7 +391,7 @@ public function toOptimizerTypeMask(): string {
             case "iterable":
```

Comment:
(GitHub doesn't allow commenting on the above line for `case "callable":`)

Does callable also need to include MAY_BE_ARRAY_OF_REF? If I use `callable` as a parameter type hint, I can still modify fields of an array callable by reference (and see them get modified)

```
php > class B { public function main() { echo "in main\n"; } }
php > $class = new B();
php > $method = 'main';
php > $callable = [&$class, &$method];
php > $callable();
in main
php > function example(callable $c) { $c(); $c[1] = 'changed'; }
php > example($callable);
in main
php > var_dump($callable);
array(2) {
  [0]=>
  &object(B)#1 (0) {
  }
  [1]=>
  &string(7) "changed"
}
php > 
```

Not sure what the other implications of this would be for the optimizer, @nikic would know more (E.g. something with a callable type hint might easily cease to be a callable?)